db-service/test/cqn4sql/flattening.test.js

```diff
@@ -390,16 +417,20 @@ describe('Flattening', () => {
               SELECT Authors.address_street, Authors.address_city from bookshop.Authors as Authors where Authors.ID > Books.ID
             )`)
     })
-    it('rejects struct fields in expressions in WHERE clause (1)', () => {
+    it('rejects struct fields in expressions in toplevel WHERE clause', () => {
       expect(() => cqn4sql(cds.ql`SELECT from bookshop.Bar { ID } WHERE 2 = nested`, model)).to.throw(
-        /Can't compare structure "nested" to value "2"; only possible for structures with one sub-element/
+        /Can't compare structure "nested" to value "2"; only possible for structures with one sub-element/,
       )
     })
 
-    it('rejects struct fields in expressions in WHERE clause (2)', () => {
-      expect(() => cqn4sql(cds.ql`SELECT from bookshop.Bar { ID } WHERE sin(nested) < 0`, model)).to.throw(
-        /A structured element can't be used as a value in an expression/,
-      )
+    it('expands struct fields in function args in WHERE clause', () => {
```

Comment:
same as sin(nested) above -> I woudn't do that

Reply:
is now rejected